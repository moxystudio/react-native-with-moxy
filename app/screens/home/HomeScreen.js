import React from 'react';
import { View, Text } from 'react-native';

import styles from './styles';

const HomeScreen = () => (
    <View style={ styles.container }>
        <Text>This is the Homescreen</Text>
    </View>
);

export default HomeScreen;
