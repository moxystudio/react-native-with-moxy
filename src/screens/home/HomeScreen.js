import React, { useCallback, useMemo } from 'react';
import PropTypes from 'prop-types';
import { Text } from 'react-native';
import HomeHeader from './header';
import { Button, useTheme } from '../../shared/react';
import { SafeAreaView } from 'react-native-safe-area-context';

import buildStyles from './styles';

const HomeScreen = ({ navigation }) => {
    navigation.setOptions({
        header: () => <HomeHeader />,
    });

    const { themeStyles } = useTheme();
    const styles = useMemo(() => buildStyles(themeStyles), [themeStyles]);

    const handlePress = useCallback(() => {
        navigation.navigate('Profile', { screen: 'Profile2' });
    }, [navigation]);

    return (
        <SafeAreaView style={ styles.container }>
            <Text style={ styles.text }>This is the Homescreen</Text>
            <Button
                type="highlight"
                title="Go To Profile"
                style={ styles.button }
                onPress={ handlePress }
                textStyle={ styles.buttonText }
                underlayColor={ themeStyles.colors.terciary } />
        </SafeAreaView>
    );
};

HomeScreen.propTypes = {
    navigation: PropTypes.object.isRequired,
};

export default HomeScreen;
