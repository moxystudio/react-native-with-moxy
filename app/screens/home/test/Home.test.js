import React from 'react';
import { mount } from 'enzyme';
import HomeScreen from '../HomeScreen';

describe('Home tests', () => {
    it('should render correctly', () => {
        const tree = mount(<HomeScreen />);

        expect(tree).toMatchSnapshot();
    });
});
