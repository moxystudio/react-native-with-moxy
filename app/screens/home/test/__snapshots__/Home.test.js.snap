// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home tests should render correctly 1`] = `
<HomeScreen>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text>
        <Text>
          This is the Homescreen
        </Text>
      </Text>
    </View>
  </View>
</HomeScreen>
`;
